compiler/rustc_const_eval/src/interpret/operand.rs

```diff
@@ -587,7 +587,8 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             }
             mir::ConstantKind::Val(val, ty) => self.const_val_to_op(*val, *ty, layout),
             mir::ConstantKind::Unevaluated(uv, _) => {
-                let instance = self.resolve(uv.def, uv.substs)?;
+                let substs = self.tcx.normalize_erasing_regions(self.param_env, uv.substs);
```

Comment:
that shouldn't be necessary because the const should already be normalized here. You could add a debug assert that this normalize call would be a noop though.

Reply:
This is necessary, because we still triggered a debug assertion without it. Not needed for the backport, per se.

Reply:
oh, ok :thinking: then we should normalize further above, can keep this with a `FIXME` for now ^^

Reply:
What do you mean further above? The unevaluated constants are the only ones that need normalization, right?

Reply:
Also, to be more clear: this normalization is needed because of the const prop lint in `mir_drops_elaborated_and_const_checked`, which happens _before_ optimized MIR. Only after optimizing the MIR can we guarantee that the `RevealAll` pass has happened and that the body's constants are normalized, so any call to `resolve` before that needs to be manually normalized.

Reply:
yeah, the const prop lint should still normalize itself if it uses `Reveal::All`

Reply:
to not block anything here: can you add a FIXME and then we merge this as is for now?

Reply:
There's a FIXME now